src/chemcaption/featurize/text.py

```
@@ -88,7 +89,7 @@ def fill_template(self, template: Any, precision_type: str = "decimal") -> str:
             PROPERTY_NAME=self.completion_names,
             REPR_SYSTEM=self.representation_type,
             REPR_STRING=self.representation,
-            PROPERTY_VALUE=join_list_elements(self.completion),
+            PROPERTY_VALUE=join_list_elements2(self.smart_names, self.completion),
```

Ruff (0.15.12)

[error] Undefined name `join_list_elements2`

(F821)